user/yaml/kubestellar-server-ingress.yaml

```diff
@@ -0,0 +1,147 @@
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kubestellar
+
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: kubestellar-pvc
+  namespace: kubestellar
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubestellar-service-account
+  namespace: kubestellar
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kubestellar-role

```

Comment:
FYI: Every use of "kubestellar-role" is going to be in a place where the reader already knows that it is a role.